File: interpreter/interpreter.js
class Numero {
    constructor(valor) {
        this.valor = valor;
    }

    interpretar() {
        return this.valor;
    }
}

class Suma {
    constructor(izq, der) {
        this.izq = izq;
        this.der = der;
    }

    interpretar() {
        return this.izq.interpretar() + this.der.interpretar();
    }
}

File: interpreter/interpreter.py
class Numero:
    def __init__(self, valor):
        self.valor = valor

    def interpretar(self):
        return self.valor


class Suma:
    def __init__(self, izq, der):
        self.izq = izq
        self.der = der

    def interpretar(self):
        return self.izq.interpretar() + self.der.interpretar()
